lib/index.js

```diff
@@ -195,11 +195,17 @@ internals.mapUri = function (protocol, host, port, uri) {
                 return next(null, uri);
             }
 
-            const address = uri.replace(/{protocol}/g, request.connection.info.protocol)
+            let address = uri.replace(/{protocol}/g, request.connection.info.protocol)
                              .replace(/{host}/g, request.connection.info.host)
                              .replace(/{port}/g, request.connection.info.port)
                              .replace(/{path}/g, request.url.path);
 
+            Object.keys(request.params).forEach((key) => {
+
+                const re = new RegExp(`{${key}}`,'g');
+                address = uri.replace(re,request.params[key]);
```

Comment:
This line removes all previous address replacements, including other params replacements. It will break existing code that uses replacements with a dynamic path segment.